test/e2e/components/vl-map.js

```diff
@@ -93,16 +93,86 @@ class VlMap extends VlElement {
     }, 2000).then(() => true).catch(() => false);
   }
 
-  async clickOnCoordinates(coordinates) {
+  /**
+   * Geef de pixels voor een coördinaat op de kaart.
+   * Relatief t.o.v. de hoogte en breedte van diezelfde kaart.
+   *
+   * @param {Number[]} coordinates - coördinaat op de kaart
+   * @return {Promise<{x: number, y: number}>}
+   * @private
+   * @see {@link https://openlayers.org/en/latest/apidoc/module-ol_Map-Map.html#getPixelFromCoordinate}
+   * @see {@link https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver_exports_WebElement.html#getRect}
+   */
+  async _getPixelsFromCoordinate(coordinates = [0, 0]) {
     const pixels = await this.driver.executeScript(`return arguments[0].map.getPixelFromCoordinate(${JSON.stringify(coordinates)})`, this);
     const rect = await this.getRect();
-    await this.driver.actions().move({
-      origin: this,
+    return {
       x: Math.round(pixels[0] - (rect.width / 2)),
       y: Math.round(pixels[1] - (rect.height / 2)),
+    };
+  }
+
+  async clickOnCoordinates(coordinates) {
+    const pixels = await this._getPixelsFromCoordinate(coordinates);
+    await this.driver.actions().move({
+      origin: this,
+      x: pixels.x,
+      y: pixels.y,
     }).click().perform();
   }
 
+  /**
+   * Beweeg een punt o.b.v zijn coördinaat naar een andere plaats op de kaart.
+   *
+   * @param {Number[]} from - coördinaat om vanuit te bewegen
+   * @param {Number[]} to - coördinaat om naar te bewegen
+   * @return {Promise<void>}
+   */
+  async movePointByCoordinates(from = [0, 0], to = [0, 0]) {
+    const fromPixels = await this._getPixelsFromCoordinate(from);
+    const toPixels = await this._getPixelsFromCoordinate(to);
+
+    await this.clickOnCoordinates(from);
+    if (await this._isFirefox()) {
+      // TODO stefanborghys: 13/02/21 moving a point (without losing the cursor) seems impossible, i cannot get this to work
```

Comment:
@coemans ik zit al een geruime tijd vast met het werkende krijgen van een e2e test voor het verplaatsen van punten in Firefox. In Chrome werkt dit steeds mooi. Maar in Firefox lijkt er iets mis te gaan met de cursor tijdens het verplaatsen van een punt. Het voelt aan als een probleem tussen de browser en de modify implementatie van Open Layers. Want als ik manueel punten verplaats merk ik dat dit ook vlotter werkt in Chrome dan in Firefox. Maar met wat gepruts en geklik werkt het wel zonder noemenswaardige problemen.
Nu ik heb al zeer veel geprobeerd op meerdere niveau's, maar niets helpt. Op de voorbeelden van OL zag ik ook dat ze snapping gebruiken en heb ook dat toegepast maar zonder positief resultaat.
Kunnen we hier eventueel eens door gaan. Misschien zie jij een oplossing.

Reply:
pascal heeft hier zonet nog eens naar meegekeken en zag dat we om het modificatie proces af te sluiten nog een eindklik nodig hebben. Na dit getest te hebben bleek dit te werken zowel in chrome als firefox. Het ziet er iets raarder uit in firefox maar de testen draaien nu wel mooi consistent groen. Bij deze is dit dus opgelost.